packages/deployer/src/utils/compiler.ts

```diff
@@ -1,12 +1,107 @@
+import { AbiType, ContractAbi, MethodAbi } from '@0xproject/types';
 import { logUtils } from '@0xproject/utils';
 import * as _ from 'lodash';
 import * as path from 'path';
 import * as solc from 'solc';
 
 import { constants } from './constants';
 import { fsWrapper } from './fs_wrapper';
-import { ContractArtifact, ContractSources } from './types';
+import { ContractArtifact, ContractSources, FunctionNameToSeenCount } from './types';
 
+/**
+ * Constructs a system-wide unique identifier for a source file.
+ * @param directoryNamespace Namespace of the source file's root contract directory.
+ * @param sourceFilePath Path to a source file, relative to contractBaseDir.
+ * @return sourceFileId A system-wide unique identifier for the source file.
+ */
+export function constructUniqueSourceFileId(directoryNamespace: string, sourceFilePath: string): string {
+    let namespacePrefix = '';
```

Comment:
this can just be reduced to `const namespacePrefix = !_.isEmpty(directoryNamespace) ? `/${directoryNamespace}` : ''`